### PR TITLE
[BC-7978] Add support for subcategory filtering

### PR DESCRIPTION
**Pull request by adamrdavid**
Added subcategories to `cross_version_category_mapping`
And use them in `VRT.all_matching_categories`

Added spec for changes.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    vrt (0.4.2)
+    vrt (0.4.3)
```
Comment (adamrdavid) [inline Gemfile.lock]: should the patch version change?

Comment (maschwenk) [reply Gemfile.lock]: it shouldn't, its just that the release was botched last time

File: lib/vrt/cross_version_mapping.rb [hunk 2]
```diff
@@ -1,12 +1,15 @@
 module VRT
   module CrossVersionMapping
     # Maps new_category_id: deprecated_node_id
+    # and new_subcategory_id: deprecated_node_id
     def cross_version_category_mapping
       category_map = {}
       deprecated_node_json.each do |key, value|
         latest_version = value.keys.sort_by { |n| Gem::Version.new(n) }.last
         id = value[latest_version].split('.')[0]
+        sub_id = value[latest_version].split('.')[0..1].join('.')
```
Comment (maschwenk) [inline lib/vrt/cross_version_mapping.rb]: just split on the `.` once and then do this properly

File: Gemfile.lock (hunk 1 above)
Comment (arcwhite) [reply Gemfile.lock]: Hey! It wasn't botched.
It was "done creatively"

**Pull request merged**